src/gui/TrackContainerView.cpp

```diff
@@ -156,45 +156,48 @@ void TrackContainerView::removeTrackView( TrackView * _tv )
 
 
 
-void TrackContainerView::moveTrackViewUp( TrackView * _tv )
+void TrackContainerView::moveTrackView( TrackView * trackView, int indexTo )
 {
-	for( int i = 1; i < m_trackViews.size(); ++i )
-	{
-		TrackView * t = m_trackViews[i];
-		if( t == _tv )
-		{
-			BBTrack::swapBBTracks( t->getTrack(),
-					m_trackViews[i - 1]->getTrack() );
-			m_scrollLayout->removeWidget( t );
-			m_scrollLayout->insertWidget( i - 1, t );
-			qSwap( m_tc->m_tracks[i-1], m_tc->m_tracks[i] );
-			m_trackViews.swap( i - 1, i );
-			realignTracks();
-			break;
-		}
-	}
+	// Can't move out of bounds
+	if ( indexTo >= m_trackViews.size() || indexTo < 0 ) { return; }
+
+	// Does not need to move to itself
+	int indexFrom = m_trackViews.indexOf( trackView );
+	if ( indexFrom == indexTo ) { return; }
+
+	BBTrack::swapBBTracks( trackView->getTrack(),
```

Comment:
@diizy In particular this is the part that worries me the most: I changed the logic of all other tracks from swapping to a move action, this can however change the location of the BB track on the selector in an inconsistent way.

I did not experience problems with this on my local tests, but should be something to worry about.